apps/onboarding/src/wallet/eip1167-proxy-deployer.ts

```diff
@@ -0,0 +1,74 @@
+import { networkConfig, NetworkConfig } from "@app/utils/config/network.config"
+import { eqAddress, Err, Ok, Result } from "@celo/base"
+import { ContractKit } from "@celo/contractkit"
+import { RawTransaction } from "@celo/contractkit/lib/wrappers/MetaTransactionWallet"
+import { Inject, Injectable, Scope } from "@nestjs/common"
+import { Contract} from "web3-eth-contract"
+import { abi as ProxyCloneFactoryABI } from '../../../../libs/blockchain/contracts/ProxyCloneFactory.json'
+import { WalletNotDeployed } from "./errors"
+import { ProxyDeployer } from "./wallet.service"
+
+/*
+ * XXX: This is a low-level implementation based on the contract version in 
+ * asaj/gas-efficiency. We should replace this with using a wrapper akin
+ * to how things look in the `legacy-proxy-deployer.ts`
+ */
+
+@Injectable({
+  // RelayerProxyService is Request scoped
+  scope: Scope.REQUEST
+})
+export class EIP1167ProxyDeployer implements ProxyDeployer {
+  deployer: Contract
+  constructor(
+    @Inject(networkConfig.KEY)
+    private readonly networkCfg: NetworkConfig,
+    private readonly contractKit: ContractKit,
+  ) {
+    this.deployer = new contractKit.web3.eth.Contract(
+      ProxyCloneFactoryABI as any,
+      networkCfg.contracts.ProxyCloneFactory
+    )
+  }
+
+  async findWallet(blockNumber: number, txHash: string, deployerAddress: string): Promise<Result<string, WalletNotDeployed>> {
+    const deployer = 
+      (deployerAddress && !eqAddress(deployerAddress, this.deployer.options.address))
+      ? new this.contractKit.web3.eth.Contract(ProxyCloneFactoryABI as any, deployerAddress)
+      : this.deployer
+
+    const events = await deployer.getPastEvents(
+      'ProxyCreated',
+      {
+        fromBlock: blockNumber,
+        toBlock: blockNumber
+      }
+    )
+
+    const deployWalletLog = events.find(event => event.transactionHash === txHash)
+
+    if (deployWalletLog) {
+      return Ok(deployWalletLog.returnValues.proxy)
+    } else {
+      return Err(new WalletNotDeployed())
+    }
+  }
+
+
+  getDeployTransaction(
+    owner: string, 
+    implementation: string, 
+    initCallData: string,
+  ): {transaction: RawTransaction, deployerAddress: string} {
+    const transaction: RawTransaction = {
+      destination: this.deployer.options.address,
+      value: "0x0",
+      data: this.deployer.methods.deployV2(implementation, initCallData).encodeABI()
```

Comment:
IIRC `deployV2` does proxy ownership transfer to the proxy, which saves gas in all phases of Komenci gas optimization listed in `asaj/gas-efficiency` in `onboarding.ts`,, but particularly phases 2 and onwards. We should think harder with @jeanregisser about whether or not we want to do this in phase 1

Reply:
Yeah let's chat about this, I also wasn't sure which `deploy` variant we should use. For me this was a decision that I made just see things working e2e -- this should be a clear todo.